warning: unknown conversion type character ‘l’ in format

```
/home/deen/isos/ddnet/ddnet-master/src/game/prng.cpp: In member function ‘void CPrng::Seed(uint64*)’:
/home/deen/isos/ddnet/ddnet-master/src/game/prng.cpp:33:62: warning: unknown conversion type character ‘l’ in format [-
Wformat=]
   33 |  str_format(m_aDescription, sizeof(m_aDescription), "%s:%016llx:%016llx", NAME, aSeed[0], aSeed[1]);
      |                                                              ^
/home/deen/isos/ddnet/ddnet-master/src/game/prng.cpp:33:70: warning: unknown conversion type character ‘l’ in format [-
Wformat=]
   33 |  str_format(m_aDescription, sizeof(m_aDescription), "%s:%016llx:%016llx", NAME, aSeed[0], aSeed[1]);
      |                                                                      ^
```
```
/home/deen/isos/ddnet/ddnet-master/src/game/server/ddracecommands.cpp: In static member function ‘static void CGameContext::ConDrySave(IConsole::IResult*, void*)’:
/home/deen/isos/ddnet/ddnet-master/src/game/server/ddracecommands.cpp:740:39: warning: unknown conversion type character ‘l’ in format [-Wformat=]
  740 |  str_format(aBuf, sizeof(aBuf), "%s-%lld-%s.save", pSelf->Server()->GetMapName(), time_get(), pSelf->Server()->GetAuthName(pResult->m_ClientID));
      |                                       ^
```
Apparently https://github.com/ddnet/ddnet/pull/2317 isn't good enough to fix the warning.

--- a/src/base/system.c
+++ b/src/base/system.c
@@ -2333,13 +2333,11 @@ int str_length(const char *str)
 int str_format(char *buffer, int buffer_size, const char *format, ...)
 {
 	int ret;
-
+#if defined(CONF_FAMILY_WINDOWS)
 	va_list ap;
 	va_start(ap, format);
-#if defined(__MINGW32__)
-	ret = __mingw_vsnprintf(buffer, buffer_size, format, ap);
-#elif defined(CONF_FAMILY_WINDOWS)
 	ret = _vsnprintf(buffer, buffer_size, format, ap);
+	va_end(ap);
 
 	buffer[buffer_size-1] = 0; /* assure null termination */
 
@@ -2348,11 +2346,13 @@ int str_format(char *buffer, int buffer_size, const char *format, ...)
 	if(ret < 0)
 		ret = buffer_size - 1;
 #else
+	va_list ap;
+	va_start(ap, format);
 	ret = vsnprintf(buffer, buffer_size, format, ap);
-#endif
 	va_end(ap);
 
 	/* null termination is assured by definition of vsnprintf */
+#endif
 
 	/* a return value of buffer_size or more indicates truncated output */
 	if(ret >= buffer_size)

--- a/src/game/prng.cpp
+++ b/src/game/prng.cpp
@@ -30,7 +30,9 @@ static unsigned int RotateRight32(unsigned int x, int Shift)
 void CPrng::Seed(uint64 aSeed[2])
 {
 	m_Seeded = true;
-	str_format(m_aDescription, sizeof(m_aDescription), "%s:%016llx:%016llx", NAME, aSeed[0], aSeed[1]);
+	str_format(m_aDescription, sizeof(m_aDescription), "%s:%08x%08x:%08x%08x", NAME,
+		(unsigned)(aSeed[0] >> 32), (unsigned)aSeed[0],
+		(unsigned)(aSeed[1] >> 32), (unsigned)aSeed[1]);
 
 	m_Increment = (aSeed[1] << 1) | 1;
 	m_State = aSeed[0] + m_Increment;

--- a/src/game/server/ddracecommands.cpp
+++ b/src/game/server/ddracecommands.cpp
@@ -736,8 +736,10 @@ void CGameContext::ConDrySave(IConsole::IResult *pResult, void *pUserData)
 	if(CSaveTeam::HandleSaveError(Result, pResult->m_ClientID, pSelf))
 		return;
 
+	char aTimestamp[32];
+	str_timestamp(aTimestamp, sizeof(aTimestamp));
 	char aBuf[64];
-	str_format(aBuf, sizeof(aBuf), "%s-%lld-%s.save", pSelf->Server()->GetMapName(), time_get(), pSelf->Server()->GetAuthName(pResult->m_ClientID));
+	str_format(aBuf, sizeof(aBuf), "%s_%s_%s.save", pSelf->Server()->GetMapName(), aTimestamp, pSelf->Server()->GetAuthName(pResult->m_ClientID));
 	IOHANDLE File = pSelf->Storage()->OpenFile(aBuf, IOFLAG_WRITE, IStorage::TYPE_ALL);
 	if(!File)
 		return;

